webapp/app/forms/validators.py

```diff
@@ -20,7 +20,7 @@
 
 class DecimalOnly:
     def __call__(self, form, field):
-        if not field.data.isdecimal():
+        if field.data and not field.data.isdecimal():
```

Comment:
The StringField will now set the data to None if nothing was provided. In the old version this was an empty string. We needed to change the DecimalOnly validator because of that. Before, we would not accept an empty string as decimalOnly. However, this could have meant an implicit inputRequired. Therefore, I have changed it.

Do you think that is correct or should we test explicitly for None?

Reply:
No, I think this is great.

Reply:
That worries me a little bit, if we check somewhere for "" but not for None. This also affects the string validators, date validators and probably others where we don't explicitly validate on InputRequired or on None.

Reply:
Great point. I had another look.

**StringField**
We use the StringField only for the SteuerlotseStringField. I have checked all the validators that we use with that field. These are `InputRequired` and `validators.length`. Both can work with None.

Fields that inherit from SteuerlotseStringField are the following. The fields marked with * check if data is None in the pre_process method and change it accordingly. I would say that those do not lead to any problems. For the others I described what I checked:
- IdNrField *
- LegacyIdNrField *
- TaxNumberField *
- UnlockCodeField *
- SteuerlotseIbanField: This does not get an error because the data is changed while being processed (s. fields.py l. 116 and StringField's process_formdata). Nevertheless, I have adapted the validator ValidIBan.
- SteuerlotseNameStringField: Does not use any special validators 
- SteuerlotseNumericStringField: Uses DecimalOnly which has been adapted.

Besides the StringField we use these fields `RadioField, Field, BooleanField, DateField, SelectField, IntegerField`. I had a look into their inheritance tree and none of them inherits from the StringField.

**Our validators**
Although they should not receive None, I have changed the following validators to check for None:
- ValidUnlockCode
- ValidElsterCharacterSet
- ValidUnlockCodeCharacterSet

I would not change the following because they all work with split fields and those should already process None correctly:
- ValidTaxNumberLength
- ValidHessenTaxNumber
- ValidTaxNumber
- ValidIdNr
- IntegerLength and NoZero: Do already work with None.